### DISCUSSION TITLE
Get a route handler `name` in a template?

### DISCUSSION THREAD

**Discussion by 4l1fe** [litestar-org/litestar#2989] [Q&A] [answered]
Would be great to get an access to a route name variable in a template instead of  pass it into a context every single time.

## What i do
Handler
```python
...
@get("/login", name='login')
async def login_page(self, request: Request) -> Template:
        return Template(template_name='login.html', context=dict(route_name='login'))
...
```

Template
```
<form hx-post="{{ url_for(route_name) }}" class="space-y-4 mt-4">
```

## What would be good
Template
```
<form hx-post="{{ url_for(request.route_handler.name) }}" class="space-y-4 mt-4">
```

Is possible to get an access to?

**Comment by Alc-Alc** [comment, marked as answer]
Heya, what you propose `url_for(request.route_handler.name)` already works inside a template. `request`  is passed by default into the template context.

Here is an example that uses HTMX and Jinja as two separate routes. The assert passes, meaning it does have access and is able to get the name.

```python
from litestar import get
from litestar.contrib.htmx.request import HTMXRequest
from litestar.contrib.htmx.response import HTMXTemplate
from litestar.contrib.jinja import JinjaTemplateEngine
from litestar.response import Template
from litestar.template.config import TemplateConfig
from litestar.testing import create_test_client


@get("hello_htmx", name="hello_htmx_route")
async def hello_htmx() -> HTMXTemplate:
    return HTMXTemplate(
        template_str="url is {{url_for(request.route_handler.name)}}", push_url="/form"
    )


@get("hello", name="hello_route")
async def hello() -> Template:
    return Template(template_str="url is {{url_for(request.route_handler.name)}}")


with create_test_client(
    [hello_htmx],
    request_class=HTMXRequest,
    template_config=TemplateConfig(directory=".", engine=JinjaTemplateEngine),
) as client:
    response = client.get("hello_htmx")
    assert response.text == "url is /hello_htmx"


with create_test_client(
    [hello],
    template_config=TemplateConfig(directory=".", engine=JinjaTemplateEngine),
) as client:
    response = client.get("hello")
    assert response.text == "url is /hello"
```

Now, if your question is to have a default name to the router itself, then it is an ongoing issue discussed over at https://github.com/litestar-org/litestar/issues/2892

**Reply by 4l1fe** [reply]
That's great! Appreciate such the comprehensive example.

Could you point me out to the documentation where `request.route_handler` is mentioned?

> Now, if your question is to have a default name to the router itself, then it is an ongoing issue discussed over at https://github.com/litestar-org/litestar/issues/2892

Yeah, before making the question i read that issue, thx.

**Reply by provinzkraut** [reply]
> Could you point me out to the documentation where request.route_handler is mentioned?

You can find this in the API reference:

https://docs.litestar.dev/latest/reference/connection.html#litestar.connection.ASGIConnection.route_handler

**Comment by 4l1fe** [comment]
> > Could you point me out to the documentation where request.route_handler is mentioned?
> 
> You can find this in the API reference:
> 
> https://docs.litestar.dev/latest/reference/connection.html#litestar.connection.ASGIConnection.route_handler

Oh, i see. `Request` inherited it from `ASGIConnection`. Nice, thank you!
